### ISSUE TITLE
Generate crossplane-runtime interface method sets

### ISSUE THREAD

**Issue by negz** [crossplaneio/crossplane-runtime#38]
### What problem are you facing?

Any Crossplane API type that wishes to leverage [crossplane-runtime](https://godoc.org/github.com/crossplaneio/crossplane-runtime/pkg)'s built in `resource.ManagedReconciler` or `resource.ClaimReconciler` must satisfy one of various interfaces, e.g. `resource.Managed`, `resource.Claim`, etc. The interfaces are typically boilerplate getters and setters of fields that are embedded into each API type.

The [Services Developer Guide](https://crossplane.io/docs/v0.3/services-developer-guide.html#defining-resource-kinds) details which common types each Crossplane API type should embed (e.g. any resource claim must embed `ResourceSpec` in its `Spec` struct). It also mentions that various getters and setters must be added to satisfy the relevant interface, but glosses over exactly how to write those getters and setters in the interest of brevity.

In summary, the set of methods an API type must implement to use the crossplane-runtime interfaces are undocumented, and are repetitive boilerplate.

### How could Crossplane help solve your problem?

Crossplane could provide a utility to generate these method sets automatically. This would allow us to avoid including lengthy examples of the required methods in the developer guide. It would also have the added bonus of making it easier to refactor the codebase if and when we wanted to alter the method set required to satisfy a core interface.

**Comment by negz** [comment]
https://github.com/negz/angryjet

Here's one I prepared earlier!

Initial work to start using this tool is in https://github.com/crossplaneio/crossplane/pull/856 and https://github.com/crossplaneio/stack-gcp/pull/39. If we want to proceed with this direction we'll also need to update the other two stacks (stack-aws and stack-azure), and update the developer guide before we consider this issue fixed.

**Comment by displague** [comment]
@negz pointed out that https://github.com/negz/angryjet should be moved to github.com/crossplaneio, potentially under a project github.com/crossplaneio/crossplane-tools.  This sounds like a good idea if only because it means we would lay the ground work for new-tool-projects once and potentially not need to do so again.  I'm all for keeping the angryjet package and/or command name within that repo.

**Comment by negz** [comment]
> I'm all for keeping the angryjet package and/or command name within that repo.

I like this compromise between sensible repo name and awesome binary name. ;)

**Comment by negz** [comment]
PRs for the other stacks:

https://github.com/crossplaneio/stack-azure/pull/26
https://github.com/crossplaneio/stack-aws/pull/29

`angryjet` now lives in https://github.com/crossplaneio/crossplane-tools/ per @displague's suggestion.

**Comment by negz** [comment]
I'm going to keep this open until I've also updated the Services Developer Guide to mention the new tools.
